kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/xstream/XStreamMarshaller.java

```diff
@@ -220,12 +251,21 @@ public String marshall(Object objectInput) {
     @Override
     public <T> T unmarshall(String input,
                             Class<T> type) {
+        xstreamContext.get().setIgnoreUnknownElements(false);
+        return (T) xstream.fromXML(input);
+    }
+
+    @Override
+    public <T> T unmarshall(String input, Class<T> type, Map<String, Object> parameters) {
+        Object ignoreString = parameters.get(XSTREAM_IGNORE_UNKNOWN);
+        xstreamContext.get().setIgnoreUnknownElements(ignoreString != null && Boolean.parseBoolean(ignoreString
+                .toString()));
         return (T) xstream.fromXML(input);
```

Comment:
reset in a finally block the flag  of ignore elements

Reply:
removing a boolean is equivalent to setting it to false, since this is set before each unmarshall, is it really worthing to include a finally block to reset if to false?
I do not think so, but done anyway.